design/ceph/ceph-nvmeof-gateway.md

```diff
@@ -0,0 +1,122 @@
+# Ceph NVMe-oF Gateway Support
+
+## Overview
+
+This design document proposes adding NVMe over Fabrics (NVMe-oF) gateway support to Rook Ceph, enabling RBD volumes to be exposed and accessed from outside the Kubernetes cluster via the NVMe/TCP protocol. 
+
+Currently, Rook Ceph provides excellent support for RBD disks through CSI drivers, but lacks a mechanism to expose RBD volumes to clients outside the Kubernetes cluster. With Ceph's deprecation of iSCSI gateway support and the introduction of NVMe-oF gateway functionality, there is an opportunity to provide external block storage access through the use of NVMeOF protocol.
+
+### Goals
+
+- Enable RBD volumes to be accessible from outside the Kubernetes cluster via NVMe/TCP protocol using rook ceph operator
+
+## Proposal:
+
+Rook will have CephNVMeOFGateway CRD for handling the communication b/w Block PVC in the k8s cluster and the client. 
+
+## Prerequisite
+
+- **Ceph Version**: Tentacle (v20) or later with NVMe-oF gateway support
+- **RBD Pools and PVCs configured**: Pre-configured RBD pools with appropriate replication/erasure coding and Existing RBD storage classes for volume provisioning.
+
+## CephNVMeOFGateway CRD
+
+### Gateway Configuration
+
+- **Group**: `gateway-group-1` - Logical grouping of gateways
+- **Replicas**: `2` - Two gateway instances for HA
+- **Port**: `5500` - NVMe-oF service port
+
+### Placement Strategy
+
+```yaml
+
+yaml
+nodeAffinity:
+  requiredDuringSchedulingIgnoredDuringExecution:
+    nodeSelectorTerms:
+    - matchExpressions:
+      - key: role
+        operator: In
+        values: ["osd-node"]
+
+```
+
+- Ensures gateways run only on nodes labeled `role=osd-node`
+- Co-locates gateways with OSD storage for optimal performance
+
+### NVMe Subsystem Configuration
+
+- **NQN**: `nqn.2016-06.io.spdk:cnode1` - Unique subsystem identifier
+- **Host Access**: Restricted to specific client NQN
+- **Security**: `allowAnyHost: false` - Only authorized hosts can connect
+
+### Storage Mapping
+
+- **Backend**: RBD image `external-volume-1` from pool `replicapool`
+- **Protocol**: NVMe namespace exposed over TCP/IP fabric with port 5500
```

Comment:
Needs more explanation of how this will need to work

Reply:
This is a good point. We should include Niels's brief explanation that "NVMe namespace" refers to a block device in NVMe-oF and always refer to it as "NVMe namespace" to help design reviewers, and we should make sure to include it later in user docs as well to help users.

Similar teminology overlap has been confusing for devs/users between "S3 storage classes" and "k8s storageclasses", and between "KMS secrets" and "Kubernetes Secrets"